File: prolog/rna-transcription/rna_transcription.pl
rna_transcription([Letter_Rna|Rna], [Letter_Dna|Dna]):-
    ( Letter_Rna == 'C' ->
        Letter_Dna = 'G'
      
      ; Letter_Rna == 'G' ->
        Letter_Dna = 'C'
      
      ; Letter_Rna == 'A' ->
        Letter_Dna = 'T'
      
      ; Letter_Dna == 'U' ->
        Letter_Dna = 'A'
    )
    rna_transcription(Rna, Dna).

File: ruby/rna-transcription/rna_transcription.rb
=begin
Write your code for the 'Rna Transcription' exercise in this file. Make the tests in
`rna_transcription_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/rna-transcription` directory.
=end

class Complement
  def self.of_dna(dna)
    case dna
    when 'G'
      'C'
    when 'C'
      'G'
    when 'T'
      'A'
    when 'A'
      'U'
    else
      ''
    end
  end
end
